engine/classes/ElggFile.php

```diff
@@ -361,6 +456,32 @@ public function exists() {
 	public function setFilestore(\ElggFilestore $filestore) {
 		$this->filestore = $filestore;
 	}
+	
+	/**
+	 * 
+	 * @return \Elgg\Filesystem\Adapter\Adapter
+	 */
+	public function getDataStorage() {
```

Comment:
This getter feels a bit spooky. Has side effects on global state?

Why do users of ElggFile need to use it to get a data storage adapter?